### DISCUSSION TITLE
Spawning Object According to 'map' frame.

### DISCUSSION THREAD

**Discussion by Kai-X-Org** [HumanSupportRobot/QandA#64] [Q&A]
For spawning the HSR using the hsrb_megaweb_2015_world.launch file, the 'map' frame has origin at (-1.0, -7.5, 0.32) in Gazebo coordinates, and this is called the 'map' frame. However, if I try to spawn objects at the orogin based on the 'map' frame directly through Gazebo, it spawns it at (0, 0, 0) in the Gazebo coordinates. What is going on? Thank you so much!

**Comment by masudam** [comment]
Since it's unclear what code was executed for spawn, this is a conjecture. The map frame is the reference for tf, and it points to (0, 0, 0) in that frame, not in Gazebo coordinates. Could it be that the values were directly used for spawning the object in Gazebo without any transformation, resulting in the object being generated at the point (0, 0, 0)?
